pkg/vault/contracts/authorizer/TimelockAuthorizer.sol

```diff
@@ -562,15 +565,21 @@ contract TimelockAuthorizer is IAuthorizer, IAuthentication, ReentrancyGuard {
         require(!scheduledExecution.cancelled, "ACTION_ALREADY_CANCELLED");
 
         // solhint-disable-next-line not-rely-on-time
-        require(block.timestamp >= scheduledExecution.executableAt, "ACTION_NOT_EXECUTABLE");
+        require(block.timestamp >= scheduledExecution.executableAt, "ACTION_NOT_YET_EXECUTABLE");
+
         if (scheduledExecution.protected) {
-            bytes32 executeScheduledActionId = getExecuteExecutionActionId(scheduledExecutionId);
-            bool isAllowed = hasPermission(executeScheduledActionId, msg.sender, address(this));
-            _require(isAllowed, Errors.SENDER_NOT_ALLOWED);
+            // Protected scheduled executions can only be executed by a set of accounts designated by the original
+            // scheduler.
+            _require(_isExecutor[scheduledExecutionId][msg.sender], Errors.SENDER_NOT_ALLOWED);
```

Comment:
Would it make sense to have a public getter for `isExecutor(id, account)`? It might make things a bit easier to read.

Reply:
Good call, yes - otherwise this information would only be available via events.